File: Selenium/Activities/Activity4.py
from selenium import webdriver
from selenium.webbrowser.common.by import By
from selenium.webbrowser.support.color import Color

with webbrowser.Firefox() as browser:
    browser.get("https://training-support.net/webelements/target-practice")
    print(f"Current page title: {browser.title}")

    third_header = browser.find_element(By.XPATH ,"//h3[contains(text(),'#3')]")
    print(f"Third header text: {third_header.text}");

    fifth_header = browser.find_element(By.XPATH,"//h5[contains(text(),'#5')]")
    header_color = Color.from_string(fifth_header.value_of_css_property("color")).rgb 
    print(f"Fifth header color: {headerColor}")

    purple_button = browser.find_element(By.XPATH,"//button[text() = 'Purple']");
    print(f"Purple button classes: {purple_button.get_attribute("class")} ")
			
    slate_button = browser.find_element(By.XPATH,"//button[text() = 'Slate']");
    print(f"Slate button text: {slate_button.text}")
    browser.quit();

File: Selenium/Activities/Activity4.java
package examples;

import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.Color;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class Activity4 {

	public static void main(String[] args) {

			WebDriver browser = new FirefoxDriver();
			browser.get("https://training-support.net/webelements/target-practice");
			System.out.println("Current page title: "+ browser.getTitle());
			
			WebElement thirdHeader = browser.findElement(By.xpath("//h3[contains(text(),'#3')]"));
			System.out.println("Third header text: " + thirdHeader.getText());
			
			WebElement fifthHeader = browser.findElement(By.xpath("//h5[contains(text(),'#5')]"));
			Color headerColor = Color.fromString(fifthHeader.getCssValue("color"));
			
			System.out.println("Fifth header color: " + headerColor.asRgb());
			
			
			WebElement purpleButton = browser.findElement(By.xpath("//button[text() = 'Purple']"));
			System.out.println("Purple button classes: " + purpleButton.getAttribute("class"));
			
			WebElement slateButton = browser.findElement(By.xpath("//button[text() = 'Slate']"));
			System.out.println("Slate button text: " + slateButton.getText());
			browser.quit();

	}

}
